Core/src/Surfaces/AnnulusBounds.cpp

```diff
@@ -100,41 +99,47 @@ std::vector<Acts::Vector2D> Acts::AnnulusBounds::corners() const {
 
 std::vector<Acts::Vector2D> Acts::AnnulusBounds::vertices(
     unsigned int lseg) const {
-  // List of vertices counter-clockwise starting with left inner
-  std::vector<Acts::Vector2D> rvertices;
-
-  double phiMinInner = VectorHelpers::phi(m_inLeftStripXY);
-  double phiMaxInner = VectorHelpers::phi(m_inRightStripXY);
-  double phiMinOuter = VectorHelpers::phi(m_outRightStripXY);
-  double phiMaxOuter = VectorHelpers::phi(m_outLeftStripXY);
-
-  std::vector<double> phisInner =
-      detail::VerticesHelper::phiSegments(phiMinInner, phiMaxInner);
-  std::vector<double> phisOuter =
-      detail::VerticesHelper::phiSegments(phiMinOuter, phiMaxOuter);
-
-  // Inner bow from phi_min -> phi_max
-  for (unsigned int iseg = 0; iseg < phisInner.size() - 1; ++iseg) {
-    int addon = (iseg == phisInner.size() - 2) ? 1 : 0;
-    detail::VerticesHelper::createSegment<Vector2D, Transform2D>(
-        rvertices, {get(eMinR), get(eMinR)}, phisInner[iseg],
-        phisInner[iseg + 1], lseg, addon);
-  }
-  // Upper bow from phi_min -> phi_max
-  for (unsigned int iseg = 0; iseg < phisOuter.size() - 1; ++iseg) {
-    int addon = (iseg == phisOuter.size() - 2) ? 1 : 0;
-    detail::VerticesHelper::createSegment<Vector2D, Transform2D>(
-        rvertices, {get(eMaxR), get(eMaxR)}, phisOuter[iseg],
-        phisOuter[iseg + 1], lseg, addon);
+  if (lseg > 0) {
+    // List of vertices counter-clockwise starting with left inner
+    std::vector<Acts::Vector2D> rvertices;
+
+    // Shift them to get the module phi
+    double phiMinInner = VectorHelpers::phi(m_inLeftStripXY - m_moduleOrigin);
+    double phiMaxInner = VectorHelpers::phi(m_inRightStripXY - m_moduleOrigin);
+    double phiMinOuter = VectorHelpers::phi(m_outRightStripXY - m_moduleOrigin);
+    double phiMaxOuter = VectorHelpers::phi(m_outLeftStripXY - m_moduleOrigin);
+
+    std::vector<double> phisInner =
+        detail::VerticesHelper::phiSegments(phiMinInner, phiMaxInner);
+    std::vector<double> phisOuter =
+        detail::VerticesHelper::phiSegments(phiMinOuter, phiMaxOuter);
+
+    // Inner bow from phi_min -> phi_max
```

Comment:
Since it's starting from the left inner corner, it's `phi_max->phi_min`?

Reply:
Will correct with left/right correctly.

Reply:
It actually worked by accident (at least for the obj displayer), because it has to create a full closed face.
Fixed now.